ehr/test/src/org/labkey/test/tests/onprc_ehr/ONPRC_EHRTest.java

```diff
@@ -195,15 +196,28 @@ public void testAssignmentApi() throws Exception
         projectSelect.addFilter(new Filter("protocol", protocolId));
         final Integer projectId = (Integer)projectSelect.execute(getApiHelper().getConnection(), getContainerPath()).getRows().get(0).get("project");
 
+        // Try with a row that doesn't pass validation
+        Map<String, Object> protocolCountsRow = new HashMap<>();
+        protocolCountsRow.put("protocol", protocolId);
+        protocolCountsRow.put("species", "Cynomolgus");
+        protocolCountsRow.put("allowed", 2);
+        protocolCountsRow.put("start", prepareDate(new Date(), -10, 0));
+        try
+        {
+            InsertRowsCommand protocolCountsCommand = new InsertRowsCommand("ehr", "protocol_counts");
+            protocolCountsCommand.addRow(protocolCountsRow);
+            protocolCountsCommand.execute(getApiHelper().getConnection(), getContainerPath());
+            fail("Should have gotten an exception due to missing end date");
+        }
+        catch (CommandException e)
+        {
+            assertEquals("Wrong failure message", "ERROR: Must enter Start and End dates", e.getMessage());
+        }
+
+        // Then try again with the problem corrected
+        protocolCountsRow.put("end", prepareDate(new Date(), 370, 0));
```

Comment:
after reading the rationale I am confused if column name here should this be "enddate" now?

Reply:
Yes, it should. I pushed and update and queued a test run:

https://teamcity.labkey.org/buildConfiguration/LabKey_2011Release_Premium_Ehr_OnprcEhrSqlserver?branch=fixAssignmentTest&mode=builds